Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorTab.cpp

```diff
@@ -124,7 +125,28 @@ namespace AzToolsFramework
                                 AZStd::move(AZStd::string::format("Could not check out asset file: %s.", assetFullPath.c_str()));
                             assetCheckoutAndSaveCallback(false, error, assetFullPath);
                         }
+                        opComplete = true;
                     });
+
+                // Broadcast is asynchronous above, (it queues an operation and returns immediately),
+                // but hooks into local variables and other lambdas that might be destroyed if we exit scope here.
+                // This operation also happens while the dialog is closing, and WILL close, destroying
+                // internal buffers like the save data itself, before this async operation proceeds and calls the callback.
+                // For these reasons we MUST block here, before proceeding.
+                //
+                // The above broadcast API also ALWAYS calls the callback, even if it failed or is unavailable, so we can bank on
+                // opComplete becoming true.  It also always calls it from the Tickbus, which is always the main GUI thread.
+
+                while (!opComplete)
```

Comment:
Could there be a situation where the callback above not execute from the `RequestEdit` call? Maybe we can add a sleep + timeout logic into this loop to prevent any possible infinite loops

Reply:
The API of the call requires that it always returns (with a failure code) and it has its own internal timeouts and stuff.  Even if P4 is active and the server goes down...

If P4 is not active, the call always returns immediately.